### ISSUE TITLE
Why xml2js? And other questions

### ISSUE THREAD

**Issue by dxg** [fredwu/spiky_xml#1]
1. I'm curious about your choice of libs.. howcome xml2json?
   `nori` uses nokogiri which has C extensions, whereas `xml2js` is pure javascript, hence it's only slightly faster than the ruby version ;)
2. Why are you measuring file read time?
   The benchmarking should begin after all the files have been read.
3. Why no concurrency?
   There is a 'concurrent' ruby version, however celluloid crashes on my VM :-\
   `ERROR -- : Celluloid::PoolManager crashed!
   ArgumentError: minimum pool size is 2`
   I'm working on an 'improved' fully concurrent node version.. i think you are too

**Comment by dxg** [comment]
Hmm `threads-a-gogo` is somewhat crippled..
Seems the only real solution would be `libuv` and a native xml parser.

**Comment by fredwu** [comment]
1. Good call, I will change it to a native version, any suggestions?
2. When read from web service, there is the file loading/parsing time too, so I'd rather keep it.
3. Is that on MRI? If so, well... imo at this point jruby is still a much better / more reliable solution for concurrency code.

**Comment by dxg** [comment]
1. xml2json which is based on libexpat
2. If you want to include IO in the benchmark then the streaming nature of http connections should be emulated;
   You would need to set up a simple web server which would transmit the xml files at internet speeds.
   Right now, reading from file (including the fact that the OS most likely caches the files after the first read) is probably akin to reading from memory _after_ the first iteration anyway.
3. Yeah, MRI.

Lastly, the way you're including node packages is a bit quirky;
The [normal method](https://github.com/buglabs/node-xml2json/blob/master/package.json) is much like ruby's Gemfile. Repo is a bit messy atm and commits are hard to follow, and I get strange errors when I clone this repo because you've also comitted the compiled binaries which don't work on my machine.

**Comment by fredwu** [comment]
Yeah I am aware of grunt and `package.json`, just haven't been bothered with this simple script. :P
